### PR TITLE
Fix binary artifacts for Mac

### PR DESCRIPTION
**Pull request by lispyclouds**
Addresses #19

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -135,7 +134,7 @@ jobs:
           name: Build binary
           command: |
             source /Users/$(whoami)/.bashrc
-            GOARCH=arm64 go build -tags "fts5" -o pod-babashka-go-sqlite3 main.go
+            go build -tags "fts5" -o pod-babashka-go-sqlite3 main.go
```
Comment (borkdude) [inline .circleci/config.yml:137]: Maybe be explicit about the environment here and keep GOARCH?

File: .circleci/config.yml [hunk 2]
```diff
@@ -50,10 +50,6 @@ jobs:
           command: curl -L https://git.io/vQhTU | bash
       - run:
           name: Build binary
-          environment:
```
Comment (borkdude) [inline .circleci/config.yml:53]: Why is this removed?

**Comment by borkdude** [comment]
@lispyclouds I made some comments but maybe you can't see them so I'm posting it as screenshot here as well.

Screenshot 2024-12-28 at 22 35 41

Screenshot 2024-12-28 at 22 35 59



It seems that the only changes needed are to the test build script + test? E.g. why is CGO removed from the linux aarch build?

File: .circleci/config.yml (hunk 2 above)
Comment (lispyclouds) [reply .circleci/config.yml:53]: I should add the CGO_ENABLED one, but the GOARCH and GOOS felt like another point of change apart from the instance type its running on. Felt better to remove since we have dedicated instances; would be better to keep if we cross compiled for all in a single instance.

File: .circleci/config.yml (hunk 1 above)
Comment (lispyclouds) [reply .circleci/config.yml:137]: Same as the other comment, thought to keep the var set only where its needed and only vary the instance types.

**Comment by lispyclouds** [comment]
Answered them, adding the CGO back but want to know of what you think of my reasons for removing the others.

**Comment by borkdude** [comment]
@lispyclouds I think we should only change something if we are solving a problem. Just changing the test binary name solves the problem so I'd only change that.

**Comment by borkdude** [comment]
Why are we even building a separate test binary?

**Comment by lispyclouds** [comment]
IIRC it was to aid in local testing, to not build separately too.

**Comment by lispyclouds** [comment]
only the file changes are there now. we could use the binary built in the prev step for the tests but for arm macs, its going to run the amd binary and not sure if rosetta is setup? or if thats a comprehensive test or not with CGO etc

**Comment by borkdude** [comment]
Installing rosetta is very simple:

```
      - run:
          name: Install Rosetta
          command: |
            sudo /usr/sbin/softwareupdate --install-rosetta --agree-to-license
```

Since rosetta wasn't a problem before, why was a separate test binary used before? I don't get the reason for that still.

**Comment by lispyclouds** [comment]
Yeah I think we can avoid the test build but like I said before, I think you added it in the shape you had it in your local test setup in https://github.com/babashka/pod-babashka-go-sqlite3/commit/f7f8ab2256bfae07b89c8916b090d421864f6cec
Having the script like this allows for a faster test cycle without needing to do a go build before testing for every code change

**Comment by borkdude** [comment]
Alright. Thanks for hunting the bug down! I'll just merge this and make a release tomorrow.
